### PR TITLE
Backlog items

### PR DESCRIPTION
**Pull request by gcapalbo**
Lots of miscellaneous fixes / tweaks / items from my backlog. Each one is a separate commit.

@millerdev @gbova

### CODE REVIEW

File: corehq/apps/sms/api.py [hunk 1]
```diff
@@ -125,6 +126,20 @@ def send_sms(domain, contact, phone_number, text, metadata=None):
...
+
+    if domain and contact and is_commcarecase(contact):
+        backend_name = contact.get_case_property('contact_backend_id')
+        backend_name = backend_name.strip() if isinstance(backend_name, six.string_types) else ''
+        if backend_name:
+            try:
+                backend = SQLMobileBackend.load_by_name(SQLMobileBackend.SMS, domain, backend_name)
+            except BadSMSConfigException as e:
+                if logged_subevent:
+                    logged_subevent.error(MessagingEvent.ERROR_GATEWAY_NOT_FOUND, additional_error_text=e.message)
```
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/sms/api.py]: `six.text_type(e)`

File: settings.py [hunk 2]
```diff
@@ -593,6 +593,9 @@
 
 # Setting this to False will make the system process outgoing and incoming SMS
 # immediately rather than use the queue.
+# This should always be set to True in production environments, and the sms_queue
+# celery worker(s) should be deployed. We set this to False for tests and (optionally)
+# for local testing.
 SMS_QUEUE_ENABLED = False
```
Comment (millerdev) [inline settings.py]: Should this be set to `True` here and `False` in [testsettings.py](https://github.com/dimagi/commcare-hq/blob/1d8bea20957f0cc97280a5857d095a558b8fa8b0/testsettings.py) and [dev_settings.py](https://github.com/dimagi/commcare-hq/blob/1d8bea20957f0cc97280a5857d095a558b8fa8b0/dev_settings.py) with a comment in dev_settings about why you might want to change it to `True` in your localsettings.py? That way production environments would automatically default to `True`.

**Review by millerdev — APPROVED** [review]
Thanks for wrapping up the loose ends.

File: settings.py (hunk 2 above)
Comment (gcapalbo) [reply settings.py]: agree, updated

**Pull request merged**